app/models/link.rb

```diff
@@ -0,0 +1,19 @@
+class Link
+  include Mongoid::Document
+  include Mongoid::Timestamps
+
+  embedded_in :link_check_report
+
+  field :uri, type: String
```

Comment:
Do we need to add some `null: false` on these?

Reply:
Will that then mean the key does not exist in mongo? If that's preferred then okay we can add to any fields we expect to be null.

Reply:
I think though that :uri will never be null as we have validations below to ensure that.

Reply:
Hmm, no, I thought it would add some validation to make sure the fields aren't ever `nil`, but it looks like you can't actually do that in Mongoid anyway.